ChainRouter.php

```diff
@@ -291,8 +291,19 @@ private function rebuildRequest($uri)
                 $server['HTTP_HOST'] .= ':'.$this->context->getHttpPort();
             }
         }
+        // use the array field to have both domain and port.
+        if (isset($server['HTTP_HOST'])) {
+            $pathinfo = $server['HTTP_HOST'].$pathinfo;
+        } else {
+            $pathinfo = 'localhost'.$pathinfo;
```

Comment:
I hope the question is not on me. I cannot comment on the code because I do not understand it.
I try to keep away from anything that comes into contact with `preparePathInfo()` and `prepareBaseUrl()` methods of `Request`, because it's a black magic.